src/formats/apple-notes.ts

```diff
@@ -365,13 +366,21 @@ export class AppleNotesImporter extends FormatImporter {
 		return Math.floor((timestamp + CORETIME_OFFSET) * 1000);
 	}
 	
-	getAttachmentPath(): string {
-		let attachmentPath = this.app.vault.getConfig('attachmentFolderPath');
-		if (attachmentPath.startsWith('/')) attachmentPath = attachmentPath.substring(1);
+	async getAttachmentPath(note: TFile): Promise<string> {
+		if (this.cachedAttachmentPath) return this.cachedAttachmentPath;
+		
+		let attachmentSetting = this.app.vault.getConfig('attachmentFolderPath');
+		let attachmentPath = parseFilePath(
+			//@ts-ignore
+			await this.app.vault.getAvailablePathForAttachments(note.basename, note.extension, note)
+		).parent;
 		
-		const outPath = path.join(attachmentPath, `${this.outputLocation} Attachments`);
+		if (!attachmentSetting.startsWith('./')) {
```

Comment:
I think we can use `path.isAbsolute` here. Be sure to import path from `../filesystem`.

Reply:
That won't work, since Obsidian's 'attachmentFolderPath' setting stores non-relative paths without the leading slash. So both will return `false` when passed into `path.isAbsolute`.

Also, path is already imported.